src/StorageSync/StorageSync/StorageSyncService/SetStorageSyncServiceIdentityCommand.cs

```diff
@@ -0,0 +1,297 @@
+                    var resourceIdentifier = new ResourceIdentifier(ResourceId);
+                    resourceName = resourceIdentifier.ResourceName;
+                    resourceGroupName = resourceIdentifier.ResourceGroupName;
+                }
+                else if (this.IsParameterBound(c => c.InputObject))
+                {
+                    resourceName = InputObject.StorageSyncServiceName;
+                    resourceGroupName = InputObject.ResourceGroupName;
+                }
+                else
+                {
+                    resourceName = Name;
+                    resourceGroupName = ResourceGroupName;
+                }
+
+
+                Target = string.Join("/", resourceGroupName, resourceName);
+                if (ShouldProcess(Target, ActionMessage))
+                {
+                    StorageSyncModels.StorageSyncService storageSyncService = StorageSyncClientWrapper.StorageSyncManagementClient.StorageSyncServices.Get(resourceGroupName, resourceName);
+                    // 1. Check if any server available for migration
+                    IEnumerable<StorageSyncModels.RegisteredServer> registeredServers = StorageSyncClientWrapper.StorageSyncManagementClient.RegisteredServers.ListByStorageSyncService(resourceGroupName, resourceName);
+                    var candidateServersLookup = new Dictionary<string, StorageSyncModels.RegisteredServer>(StringComparer.InvariantCultureIgnoreCase);
+                    var clusterNameServersLookup = new Dictionary<string, StorageSyncModels.RegisteredServer>(StringComparer.InvariantCultureIgnoreCase);
+                    foreach (var registeredServer in registeredServers)
+                    {
+                        // Scenario : Server is running in certificate mode
+                        if((registeredServer.ActiveAuthType == StorageSyncModels.ServerAuthType.Certificate && !string.IsNullOrEmpty(registeredServer.ApplicationId) && registeredServer.ApplicationId != Guid.Empty.ToString())
+                            || (Guid.TryParse(registeredServer.LatestApplicationId, out Guid latestApplicationId)))
+                        {
+                            StorageSyncClientWrapper.VerboseLogger.Invoke($"Server :  {registeredServer.ServerName ?? registeredServer.ServerId} ({registeredServer.ServerRole})");
+                            candidateServersLookup.Add(registeredServer.Id, registeredServer);
+
+                            if(registeredServer.ServerRole == ServerRoleType.ClusterNode.ToString())
+                            {
+                                var clusterNameServer = registeredServers.SingleOrDefault(s => s.ServerId == registeredServer.ClusterId);
+                                if(clusterNameServer != null)
+                                {
+                                    throw new PSArgumentException($"Cluster Name Server {clusterNameServer.ServerName} is not found for Cluster Node {registeredServer.ServerName}");
```

Comment:
should this ever happen?

Reply:
Client cannot make cloud assumptions or assert